stdlib/universe/union2.go

```diff
@@ -31,6 +34,9 @@ func (u *unionTransformation2) ProcessMessage(m execute.Message) error {
 		return nil
 	}
 
+	u.mu.Lock()
```

Comment:
Can you add a comment for why a mutex is needed?

Reply:
This was caught by `test-race` which is how I know about it. It doesn't seem to be the most consistent though. Either way, I think that does mean there's a test case for this.

Reply:
To expand on this request a bit, I am concerned that having a mutex within the Process message code is an anti pattern and so I want to understand why its needed as I would expect it to be the exception rather than the rule.

Reply:
Oh whoops. I was responding to your overall comment and it looks like I'm responding to the request for a comment. I'm adding a comment. I do think it would probably be good to fix this in the engine to remove these needed locks. I don't think there's a time when the lock is not needed for concurrent parents so we might just want to do it automatically.

Reply:
I added the comment. Not sure how to add another test since there's already a test that uses union with two parents in a concurrent way.

Reply:
Sounds good, so generally the Process method doesn't need to lock but only because we have multiple predecessors in the case of union we need to. Makes sense